### PR TITLE
aws - efs - add has-statement filter

### PR DESCRIPTION
**Pull request by StevenGunn**
This adds support to filter on AWS EFS file system policies:

```
    filters:
      - type: has-statement
        statements:
          - Effect: "Allow"
            Condition:
              Bool:
                "elasticfilesystem:AccessedViaMountTarget": "true"
            Resource: "{fs_arn}"
```

### CODE REVIEW

**Comment by StevenGunn** [comment]
@thisisshi can you check this one out when you get a chance?

File: c7n/resources/efs.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from c7n.filters import Filter
 from c7n.manager import resources
 from c7n.filters.vpc import SecurityGroupFilter, SubnetFilter
+import c7n.filters.policystatement as polstmt_filter
```
Comment (thisisshi) [inline c7n/resources/efs.py:11]:
```suggestion
from c7n.filters.policystatement import HasStatementFilter
```

File: c7n/resources/efs.py [hunk 2]
```diff
@@ -304,3 +305,35 @@ def process(self, resources, event=None):
             "%d of %d EFS policies don't enforce secure transport",
             len(results), len(resources))
         return results
+
+
+@ElasticFileSystem.filter_registry.register('has-statement')
+class HasStatementFilter(polstmt_filter.HasStatementFilter):
```
Comment (thisisshi) [inline c7n/resources/efs.py:311]:
```suggestion
class EFSHasStatementFilter(HasStatementFilter):
```

**Review by thisisshi — CHANGES_REQUESTED** [review]

**Comment by StevenGunn** [comment]
@thisisshi requested changes have ben made, can you please review/approve when you can?

**Review by thisisshi — APPROVED** [review]

**Pull request merged**